rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -1558,9 +1626,14 @@ private TransactionGateway getTransactionGateway() {
 
     private NodeMessageHandler getNodeMessageHandler() {
         if (nodeMessageHandler == null) {
+            RskSystemProperties rskSystemProperties = getRskSystemProperties();
+            BlockProcessor blockProcessor = rskSystemProperties.fastBlockPropagation()
```

Comment:
Why not to delegate this logic to a block processor getter? Or is there a situation when even having fast block propagation enable, we should need the direct block processor instead of async one?

Reply:
done